packaging/os/pulp_repo.py

```diff
@@ -0,0 +1,509 @@
+            export_distributor["distributor_config"] = dict()
+            export_distributor["distributor_config"]["http"] = serve_http
+            export_distributor["distributor_config"]["https"] = serve_https
+            export_distributor["distributor_config"]["relative_url"] = relative_url
+            data["distributors"].append(export_distributor)
+
+            data["importer_type_id"] = "yum_importer"
+            data["importer_config"] = dict()
+            if feed_url:
+                data["importer_config"]["feed"] = feed_url
+            if proxy_host:
+                data["importer_config"]["proxy_host"] = proxy_host
+            if proxy_port:
+                data["importer_config"]["proxy_port"] = proxy_port
+            data["notes"] = {
+                "_repo-type": "rpm-repo"
+            }
+        # self.module.fail_json(msg=data)
+        response, info = fetch_url(self.module, url, data=json.dumps(data), method="POST")
+
+        if info['status'] != 201:
+            self.module.fail_json(
+                msg="Failed to create repo.",
+                status_code=info['status'],
+                response=info['msg'],
+                url=url
+            )
+        else:
+            return True
+
+    def delete_repo(self, repo_id):
+        url = self.host + '/pulp/api/v2/repositories/%s/' % repo_id
+        response, info = fetch_url(self.module, url, data='', method="DELETE")
+
+        if info['status'] != 202:
+            self.module.fail_json(
+                msg="Failed to delete repo.",
+                status_code=info['status'],
+                response=info['msg'],
+                url=url
+            )
+        return True
+
+    def get_repo_config_by_id(self, repo_id):
+        if repo_id not in self.repo_cache.keys():
+            repo_array = [x for x in self.repo_list if x["id"] == repo_id]
+            self.repo_cache[repo_id] = repo_array[0]
+
+        return self.repo_cache[repo_id]
+
+    def publish_repo(self, repo_id, publish_distributor):
+        url = self.host + '/pulp/api/v2/repositories/%s/actions/publish/' % repo_id
+        # If there's no distributor specified, we will publish them all
+        if publish_distributor is None:
+            repo_config = self.get_repo_config_by_id(repo_id)
+            for distributor in repo_config["distributors"]:
+                data = dict()
+                data["id"] = distributor["id"]
+                response, info = fetch_url(self.module, url, data=json.dumps(data), method="POST")
+                if info['status'] != 202:
+                    self.module.fail_json(
+                        msg="Failed to publish the repo the repo.",
+                        status_code=info['status'],
+                        response=info['msg'],
+                        url=url,
+                        distributor=distributor["id"]
+                    )
+        else:
+            data = dict()
+            data["id"] = publish_distributor
```

Comment:
I have noticed inconsistency in quoting across your code. There is a convention that dictionary keys should be single-quoted (e.g. `data['id'] = publish_distributor`) and text messages should be double quoted (e.g. `msg="Failed to publish the repo the repo."`). It would be nice to unify this across your code.